### PR TITLE
Fixed #30511 -- Used identity columns instead of serials on PostgreSQL.

### PR DESCRIPTION
**Pull request by apollo13**

### CODE REVIEW

**Comment by apollo13** [comment]
buildbot, test on oracle

**Comment by apollo13** [comment]
buildbot, test on oracle.

File: django/db/backends/postgresql/schema.py [hunk 1]
```diff
@@ -118,48 +126,31 @@ def _alter_column_type_sql(self, model, old_field, new_field, new_type):
...
+                        self.sql_add_identity
                         % {
                             "table": self.quote_name(table),
-                            "changes": self.sql_alter_column_default
-                            % {
-                                "column": self.quote_name(column),
-                                "default": "nextval('%s')"
-                                % self.quote_name(sequence_name),
-                            },
+                            "column": self.quote_name(strip_quotes(old_field.column)),
```
Comment (felixxm) [inline django/db/backends/postgresql/schema.py]: Why we have here a `column` from the `old_field`?
```suggestion
                            "column": self.quote_name(column),
```

**Review by felixxm — COMMENTED** [review]
@apollo13 Thanks :+1: This looks great :1st_place_medal: I pushed small edits, added release notes, and removed some leftovers.

@timgraham Do you want to take a look? `django-cockroachdb` is based on the PostgreSQL backend, so it can be significant for you.

**Review by timgraham — APPROVED** [review]

File: docs/releases/4.1.txt [hunk 2]
```diff
@@ -309,6 +309,9 @@ Models
   allows customizing attributes of fields that don't affect a column
   definition.
 
+* On PostgreSQL, ``AutoField``, ``BigAutoField``, and ``SmallAutoField`` are
+  now created as identity columns.
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/releases/4.1.txt]: There are advantages of using identity columns e.g. we don't have to care about owners of sequences. However I'm not sure if it's worth mentioning in release notes :thinking:

**Comment by shiplu** [comment]
I already have more than 50 tables in production that use sequence type for id columns. Creating a new table now creates an id column with identity type. My question is, shouldn't there be a migration to convert the existing sequence column to identity when you upgrade to Dj 4.1? 

1. If migrations were not required, are the sequence and identity types 100% compatible? 
1. Should I migrate the existing sequence columns?

**Comment by felixxm** [comment]
> * If migrations were not required, are the serial and identity types 100% compatible?

Yes, they should be compatible.
 
> * Should I migrate the existing serial columns?

There is no need to migrate existing columns.

Please see [UseSupportChannels](https://code.djangoproject.com/wiki/TicketClosingReasons/UseSupportChannels) for ways to get help with Django usage.

**Comment by boxed** [comment]
Thanks @shiplu for making sure people who google and end up here get this valuable answer.

**Comment by ngnpope** [comment]
@shiplu While you don't need to upgrade existing tables to identity columns, if you decide you want to, see [this blog post](https://www.enterprisedb.com/blog/postgresql-10-identity-columns-explained) for a solution.

**Pull request merged**